main.py

```diff
@@ -1,80 +1,58 @@
+        photo_attributes["farm"], photo_attributes["server"], photo_attributes['id'], photo_attributes["secret"]))
+    return {"source": flickr_binary_file.content, "name": photo_attributes["title"]}
 
 
-def twitter_query(media_id):
-    # return 1
-    consumer = oauth2.Consumer(key=CONSUMER_KEY, secret=CONSUMER_SECRET)
-    access_token = oauth2.Token(key=ACCESS_KEY, secret=ACCESS_SECRET)
-    client = oauth2.Client(consumer, access_token)
-    endpoint = "https://api.twitter.com/1.1/statuses/update.json"
-    params = urlencode({'status': 'Hello from Python + Heroku',
-                        'media_ids': media_id})
-    r, c = client.request(endpoint, "POST", params)
-    print(c)
+def upload_picture_to_twitter(pic):
+    endpoint = "https://upload.twitter.com/1.1/media/upload.json"
+    auth = OAuth1(CONSUMER_KEY, client_secret=CONSUMER_SECRET, resource_owner_key=ACCESS_KEY,
+                  resource_owner_secret=ACCESS_SECRET, signature_type='body')
+    params = {'name': pic["name"], 'media_data': base64.b64encode(pic["source"])}
+    content = requests.post(endpoint, params, auth=auth)
+    return {"id": json.loads(content.text)['media_id'], "name": pic["name"]}
 
 
-def upload_picture_to_twitter(bin_pic):
-    consumer = oauth2.Consumer(key=CONSUMER_KEY, secret=CONSUMER_SECRET)
-    access_token = oauth2.Token(key=ACCESS_KEY, secret=ACCESS_SECRET)
-    client = oauth2.Client(consumer, access_token)
-    endpoint = "https://upload.twitter.com/1.1/media/upload.json"
-    b64_pic = base64.b64encode(bin_pic)
-    params = urlencode({'name': 'picture',
-                        'media_data': b64_pic})
-    status, content = client.request(endpoint, "POST", params,
-                                     urlencode({'Content-Type': 'multipart/form-data'}))
-    return json.loads(content)['media_id']
+def post_to_twitter(text_to_tweet, id_of_photo):
+    auth = OAuth1(CONSUMER_KEY, client_secret=CONSUMER_SECRET, resource_owner_key=ACCESS_KEY,
+                  resource_owner_secret=ACCESS_SECRET, signature_type='body')
+    endpoint = "https://api.twitter.com/1.1/statuses/update.json"
+    params = {'status': text_to_tweet, 'media_ids': id_of_photo}
+    c = requests.post(endpoint, params, auth=auth)
+    print(c)
 
 
 if __name__ == "__main__":
-    h = httplib2.Http(".cache")
-    pictures_list = flickr_query_to_get_list_of_pictures(h)
-    indexes = random.sample(range(0, len(pictures_list)), len(pictures_list))
-    pic = flickr_query2(h, pictures_list[0])
-    pic_id = upload_picture_to_twitter(pic)
-    twitter_query(pic_id)
-    '''for index in indexes:
-        pic = flickr_query2(h, pictures_list[index])
-        with open("pic{}.jpg".format(index), "wb") as file:
-            file.write(pic)
-    '''
+    will_be_uploaded_N_photos = 10
+    while (will_be_uploaded_N_photos>0):
+        pictures_list = flickr_get_photos_list()
+        random_index = random.choice(range(0, len(pictures_list)))
```

Comment:
Can be replaced with
random_picture = random.choice(pictures_list)
photo_flickr = flickr_get_photo(random_picture )

Reply:
I didn't understand that list IS interval, thank you!